utils/upload.go

```diff
@@ -72,6 +75,38 @@ func UploadToCloudStorage(uploader Uploader, path string) {
 	uploader.Upload(walker)
 }
 
+func GetPreSignedURL(videoId string) string {
+	cloudSession := configs.GetCloudSession()
+
+	containerName := constants.CLOUD_CONTAINER_NAME_TEMP
+
+	if configs.EnvVar[configs.GCP_PROJECT_ID] != "" {
+		return getPresignedGCPURL(&GcpUploader{
```

Comment:
Just a suggestion, correct me if I am wrong .why are we passing a pointer to the function generating predesigned  URL for GCP, shouldn't the code be uniform for every url generating function?